assets/js/components/KeyMetrics/MetricsSelectionPanel/Metrics.js

```diff
@@ -32,83 +37,143 @@ import { CORE_MODULES } from '../../../googlesitekit/modules/datastore/constants
+			return acc;
+		}
+
+		if (
+			typeof KEY_METRICS_WIDGETS[ metric ].displayInList === 'function' &&
+			! displayInList( metric )
+		) {
+			return acc;
+		}
+
+		const widget = getWidget( metric );
+
+		KEY_METRICS_WIDGETS[ metric ].disconnectedModules =
+			widget.modules.reduce( ( modulesAcc, slug ) => {
+				const module = getModule( slug );
+				if ( module?.connected || ! module?.name ) {
+					return modulesAcc;
 				}
 
-				if (
-					typeof KEY_METRICS_WIDGETS[ metric ].displayInList ===
-						'function' &&
-					! KEY_METRICS_WIDGETS[ metric ].displayInList(
-						select,
-						isViewOnlyDashboard
-					)
-				) {
-					return acc;
-				}
+				return [ ...modulesAcc, module.name ];
+			}, [] );
+
+		return { ...acc, [ metric ]: KEY_METRICS_WIDGETS[ metric ] };
+	};
 
-				const widget = select( CORE_WIDGETS ).getWidget( metric );
+	const widgets = useSelect(
+		( select ) =>
+			select( CORE_WIDGETS ).getWidgets(
+				AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY
+			) || []
+	);
 
-				KEY_METRICS_WIDGETS[ metric ].disconnectedModules =
-					widget.modules.reduce( ( modulesAcc, slug ) => {
-						const module = getModule( slug );
-						if ( module?.connected || ! module?.name ) {
-							return modulesAcc;
-						}
+	const savedMetricSlugs = widgets
+		.filter( ( { slug } ) => savedMetrics.includes( slug ) )
+		.map( ( { slug } ) => slug );
 
-						return [ ...modulesAcc, module.name ];
-					}, [] );
+	const availableSavedMetrics = Object.keys( KEY_METRICS_WIDGETS )
+		.filter( ( metricSlug ) => {
+			return savedMetricSlugs.includes( metricSlug );
+		} )
+		.reduce( metricsListReducer, {} );
 
-				return { ...acc, [ metric ]: KEY_METRICS_WIDGETS[ metric ] };
-			}, {} );
-	} );
+	const availableUnsavedMetrics = Object.keys( KEY_METRICS_WIDGETS )
+		.filter( ( metricSlug ) => {
+			return ! savedMetricSlugs.includes( metricSlug );
+		} )
+		.reduce( metricsListReducer, {} );
 
 	return (
 		<div className="googlesitekit-km-selection-panel-metrics">
-			{ Object.keys( availableMetrics ).map( ( slug ) => {
-				const { title, description, disconnectedModules } =
-					availableMetrics[ slug ];
-
-				const id = `key-metric-selection-checkbox-${ slug }`;
-
-				return (
-					<MetricItem
-						key={ id }
-						id={ id }
-						slug={ slug }
-						title={ title }
-						description={ description }
-						disconnectedModules={ disconnectedModules }
-						savedMetrics={ savedMetrics }
-					/>
-				);
-			} ) }
+			{
+				// Split list into two sections with sub-headings for current selection and
+				// additional metrics if there are already saved metrics.
+				savedMetrics.length !== 0 && (
+					<Fragment>
+						<p className="googlesitekit-km-selection-panel-metrics__subheading">
+							{ __( 'Current selection', 'google-site-kit' ) }
+						</p>
+						<div className="googlesitekit-km-selection-panel-metrics__subsection">
+							{ Object.keys( availableSavedMetrics ).map(
+								( slug ) => {
```

Comment:
Maybe this block can be made into a re-usable util function, for example `renderMetrics` that would accept the metrics array and `savedMetrics` as parameters. Then it can be used like `renderMetrics( availableSavedMetrics, savedMetrics )` and `renderMetrics( availableUnsavedMetrics, savedMetrics )` to reduce the repetition and file size.